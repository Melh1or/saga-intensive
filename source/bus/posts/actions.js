// Types
import { types } from "./types";

export const postsActions = {
    fillPosts: (posts) => {
        return {
            type:    types.FILL_POSTS,
            payload: posts,
        };
    },
    createPost: (post) => {
        return {
            type:    types.CREATE_POST,
            payload: post,
        };
    },
    createPostAsync: (comment) => {
        return {
            type:    types.CREATE_POST_ASYNC,
            payload: comment,
        };
    },
    fetchPostsAsync: () => {
        return {
            type: types.FETCH_POSTS_ASYNC,
        }
    },
    clearPosts: () => {
        return {
            type: types.CLEAR_POSTS,
        }
    },
    removePostAsync: (postId) => {
        return {
            type: types.REMOVE_POST_ASYNC,
            payload: postId
        }
    },
    removePost: (postId) => {
        return {
            type: types.REMOVE_POST,
            payload: postId
        }
    },
    likePostAsync: (postId) => {
        return {
            type: types.LIKE_POST_ASYNC,
            payload: postId
        }
    },
    unlikePostAsync: (postId) => {
        return {
            type: types.UNLIKE_POST_ASYNC,
            payload: postId
        }
    },
    likePost: (likePostData) => {
        return {
            type: types.LIKE_POST,
            payload: likePostData
        }
    },
    unlikePost: (likePostData) => {
        return {
            type: types.UNLIKE_POST,
            payload: likePostData
        }
    }
}
