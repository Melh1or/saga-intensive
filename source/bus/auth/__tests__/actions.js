// Actions
import { profileActions } from '../../profile/actions';
import { authActions } from '../actions';

// Types
import { types } from '../types';

describe('auth actions: ', () => {
    test('authenticate', () => {
        expect(authActions.authenticate()).toEqual({
            type: types.AUTHENTICATE,
        });
    });

    test('initialize', () => {
        expect(authActions.initialize()).toEqual({
            type: types.INITIALIZE,
        });
    });

    test('logout', () => {
        expect(authActions.logout()).toEqual({
            type: types.LOGOUT,
        });
    });
  
    test('signupAsync', () => {
        expect(authActions.signupAsync(__.userProfile)).toEqual({
            type:    types.SIGNUP_ASYNC,
            payload: __.userProfile,
        });
    });
  
    test('loginAsync', () => {
        expect(authActions.loginAsync(__.credentials)).toEqual({
            type:    types.LOGIN_ASYNC,
            payload: __.credentials,
        })
    })
});

describe('profile actions', () => {
    test('fillUsers', () => {
        expect(profileActions.fillProfile(__.userProfile)).toMatchSnapshot()
    })
})
