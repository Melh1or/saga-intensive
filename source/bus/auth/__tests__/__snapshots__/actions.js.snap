// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile actions fillUsers 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE",
}
`;
